### PR TITLE
Create Dhruv561.md

### PR DESCRIPTION
**Pull request by Dhruv561**
## Describe your project in 1-3 sentences
I am building a smart display which provides information such as weather, calendar whilst being able to control smart devices powered by the rasberry pi. I might turn this into a magic mirror later on.

## Hack Club Slack handle
Dhruv Verma

## Checklist before submitting proposal:

- [Yes ] I am a current high school student (or younger).
- [Yes] I have submitted an RSVP at <https://hackclub.com/winter>
- [Yes] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [Yes] I have self-reviewed my project proposal.

### CODE REVIEW

**Comment by Dhruv561** [comment]
---
 name: @Dhruv561
 project: Arduino Robots
 ---

 # Project Name
 Arduino Robots

 ## Summary

 I’m going to be building a robotic hexapod and car which are controlled using an Arduino board that I can have fun building. The goal of this project is to create something that is super cool that I can control using a remote to move. I am also going to build an Arduino controlled smart lamp which I would use in my bedroom.



 ## Plan

 I’ll be ordering an Arduino uno kit off of Amazon which will give me all the tools and equipment needed for building all the projects. Then I will order the robot kits for a hexapod and car which I will assemble and connect and control with the Arduino using the instructions as well as YouTube videos online. For the smart lamp I will be using the Yeelight and putting it in my lamp which will be wired to the Arduino allowing me to control it. I will also be following a YouTube video for this.
 https://youtu.be/nivTZeGthf4
 https://youtu.be/5b8QwkJIg5E
 https://youtu.be/BgiEOCgc5ME


 ## Budget

 | Product         | Supplier/Link                         | Cost   |
 | --------------- | ------------------------------------- | ------ |
 | Arduino Kit        | https://amzn.asia/d/7v33Fo3               | $55  |
 | Yeelight Bulb     | https://amzn.asia/d/eI2uO3t                | $30  |
 | Arduino Kit        | https://amzn.asia/d/69hXE4v               | $155 |
 | Arduino Kit        | https://amzn.asia/d/0wIgxJd                | $44  |
 | Total                   |                                                                | $244 |

**Review by bellesea — CHANGES_REQUESTED** [review]
hiii @Dhruv561 this is such a cool idea but i'm a little hesitant to approve purchasing the ready kits that you linked... what do you think about maybe selecting just one project (like the hexapod) and doing the research to figure out what parts you need, what code you need to write, and maybe certain modifications you want to make to really make the project your own?

or if you think that a kit would be the ideal, could you please expand on why?

thank you sm!

**Comment by Dhruv561** [comment]
The reason I would prefer kits is because I have almost no experience with building something from scratch like that. The benefit of the kit is that I ensure that everything I need for the build would be there. That is the risk I run without getting a kit is that if parts are not compatible with one another, the project would simply go to waste and I don’t want that. Because i have no experience with an arduino or building a robot ever before, I think dipping my toes into this area with kits is far better than going to try make it on my own and adding modifications and end up making nothing due to just not knowing how everything works. Another thing is I have no items such as soldering irons or wood saws so it is also a practical inconvenience to try build everything from random parts. As a beginner I would much rather experience some success using the kits than feel as if I just wasted time and money giving up on a project that tried to make from scratch without experience. As for the code relating to the kits, I will not simply copy it. Coding is one thing I am interested in learning and I would much rather just use trial and error than copy that code.

**Comment by bellesea** [comment]
> The reason I would prefer kits is because I have almost no experience with building something from scratch like that. The benefit of the kit is that I ensure that everything I need for the build would be there. That is the risk I run without getting a kit is that if parts are not compatible with one another, the project would simply go to waste and I don’t want that. Because i have no experience with an arduino or building a robot ever before, I think dipping my toes into this area with kits is far better than going to try make it on my own and adding modifications and end up making nothing due to just not knowing how everything works. Another thing is I have no items such as soldering irons or wood saws so it is also a practical inconvenience to try build everything from random parts. As a beginner I would much rather experience some success using the kits than feel as if I just wasted time and money giving up on a project that tried to make from scratch without experience. As for the code relating to the kits, I will not simply copy it. Coding is one thing I am interested in learning and I would much rather just use trial and error than copy that code. @bellesea

@Dhruv561 thank you for your patience with the response (we were out for christmas break). ahhh i appreciate and can relate to what you're saying! however, the purpose of this grant & event is to give you a chance to explore and learn, and there are beginner-friendly ways to do this outside of using a kit? like maybe following an open-sourced building guide?

eg. https://www.instructables.com/Hexapod-1/ or https://www.instructables.com/Arduino-Mega-Hexapod/

both would give you a base to jump off while keeping it as modifiable as you want :)

what do you think?

**Comment by Dhruv561** [comment]
Hey @bellesea after looking at some more project ideas I’ve decided to scrap the arduino kits and instead opt to build a smart home controlling digital display using the raspberry pi. We have a spare tv now and the perfect use case for it. Can you take a look at this?

File: Dhruv561.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+## Plan
+
+I'll be ordering a rasberry pi 4 off of the official rasberry pi website which will be used to power the smart display. I already have access to a large display due to having a spare tv. I will then use the rasberry pi starter kit snd accesories to wire the tv to the rasberry pi and be able to control it. The accesories kit is very beneficial here as it contains all the wires I would need to possibly use when connecting to the tv. I will then watch some youtube videos and learn how code the rasberry pi to display and sync all of my smart devices and display the vital infirmation I need using youtube videos. Throughout the whole process I will be following videos which ensure that I am building the smart display correctly. This is the intial plan however after this I may on my own purchase a two way mirror and a frame for tv and turn it into a full magic mirror powered by the rasberry pi if I have the time and budget. I would do this by placing the tv into a wooden ikea frame and then placing a two way mirror film on top to create the illusion of a smart mirror.
+https://youtu.be/4hN0_yGVdUQ
```
Comment (bellesea) [inline Dhruv561.md:16]:
```suggestion

https://youtu.be/4hN0_yGVdUQ
```

File: Dhruv561.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+https://www.cnet.com/home/smart-home/turn-an-old-monitor-into-a-wall-display-with-a-raspberry-pi/
```
Comment (bellesea) [inline Dhruv561.md:17]:
```suggestion

https://www.cnet.com/home/smart-home/turn-an-old-monitor-into-a-wall-display-with-a-raspberry-pi/
```

**Review by bellesea — CHANGES_REQUESTED** [review]
@Dhruv561 this looks awesome!! do you plan on referencing any guides of sorts when building this? also, it doesn't look like the accessories box includes a power supply and SD card which you'll need to be able to program your pi? 

and, do you need a HDMI cable to connect your pi4 to your display?

also, how do you plan on programming your pi? are you thinking of a headless set up or will you be connecting a mouse and keyboard?
https://www.tomshardware.com/reviews/raspberry-pi-headless-setup-how-to,6028.html
https://linuxhint.com/connect-raspberry-pi-monitor/

thanks sm!

**Comment by Dhruv561** [comment]
Hey @bellesea as for your hardware concerns, the sd card, adaptor, card reader, hdmi cable, etc are all included with the Rasberry pi (recheck the first link) itself so I should have everything I require in terms of hardware for this project. As for programming the pi, I am thinking of using magic mirror and following this guide and some videos based around this:
https://docs.magicmirror.builders/getting-started/installation.html#alternative-installation-methods

After setting up magic mirror I will use the modules library to add in all the different things I want my display to be able to do/show using this:
https://github.com/MichMich/MagicMirror/wiki/3rd-party-modules
https://magpi.raspberrypi.com/articles/magic-mirror-modules

I am also going to opt for a wired connection to a keyboard and mouse to start with and I might switch to a headless setup after everything works properly.

**Comment by bellesea** [comment]
> Hey @bellesea as for your hardware concerns, the sd card, adaptor, card reader, hdmi cable, etc are all included with the Rasberry pi (recheck the first link) itself so I should have everything I require in terms of hardware for this project. As for programming the pi, I am thinking of using magic mirror and following this guide and some videos based around this: https://docs.magicmirror.builders/getting-started/installation.html#alternative-installation-methods
> 
> After setting up magic mirror I will use the modules library to add in all the different things I want my display to be able to do/show using this: https://github.com/MichMich/MagicMirror/wiki/3rd-party-modules https://magpi.raspberrypi.com/articles/magic-mirror-modules
> 
> I am also going to opt for a wired connection to a keyboard and mouse to start with and I might switch to a headless setup after everything works properly.

@Dhruv561 ahh my bad, thanks for highlighting that! great, could you please update the file to include your additional information about what you want to do after setting up the magic mirror?

**Comment by Dhruv561** [comment]
@bellesea i have added some more information I hope it is enough :)

File: Dhruv561.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+https://docs.magicmirror.builders/getting-started/installation.html#alternative-installation-methods
+
+After setting up magic mirror I will use the modules library to add in all the different things I want my display to be able to do/show using this:
+https://github.com/MichMich/MagicMirror/wiki/3rd-party-modules
```
Comment (bellesea) [inline Dhruv561.md:19]:
```suggestion

- https://github.com/MichMich/MagicMirror/wiki/3rd-party-modules
```

File: Dhruv561.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+https://magpi.raspberrypi.com/articles/magic-mirror-modules
```
Comment (bellesea) [inline Dhruv561.md:20]:
```suggestion
- https://magpi.raspberrypi.com/articles/magic-mirror-modules
```

File: Dhruv561.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
+name: @Dhruv561
+project: Rasberry Pi Digital Display
+---
+
+# Project Name
+Rasberry Pi Dogital Display
```
Comment (bellesea) [inline Dhruv561.md:7]:
```suggestion
Rasberry Pi Digital Display
```

**Review by bellesea — APPROVED** [review]
@Dhruv561 this looks awesome!! can't wait to see you build this :))

an awesome member of @hackclub/bank-ops will be reaching out soon with next steps!

**Comment by Dhruv561** [comment]
Whoops @bellesea i accidentally requested for another review. My bad. Btw do you have a rough estimate for how long it will take from one of the team members to reach out in slack?

**Review by bellesea — APPROVED** [review]

**Comment by bellesea** [comment]
@Dhruv561 we're a little backlogged with projects right now so likely today or early next week! thank you so much for your patience :)

**Review by MelSmith104 — COMMENTED** [review]
hey @Dhruv561 this project looks great! i love the smart displays :) im reaching out to you on slack so that we can chat about next steps!

**Pull request merged**